### DISCUSSION TITLE
Testing rtsp stream from IP camera

### DISCUSSION THREAD

**Discussion by farshidbakh** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1976] [Q&A] [answered]
Hello All, I am new to Kinesis webrtc. I am testing with the C sdk, and the Kinesis Test Page. 
I tested with the 
kvsWebrtcClient chanelnumber1 

it works fine.. I see the stream and hear the audio on the test page.
 But when I try to send my camera's rtsp by running

kvsWebrtcClientMasterGstSample chanelnumber1 video-only rtspsrc rtsp://192.168.1.97:554/stream 

I don't get the stream on the Kinesis test page

This is the method being called inside the C code. (I added my comment below)

 Sending to gst_parse_launch********** uridecodebin uri=rtsp://192.168.1.97:554/stream ! videoconvert ! x264enc bframes=0 speed-preset=veryfast bitrate=512 byte-stream=TRUE tune=zerolatency ! video/x-h264,stream-format=byte-stream,alignment=au,profile=baseline ! queue ! appsink sync=TRUE emit-signals=TRUE name=appsink-video 

Any help would greatly be appreciated.
Thanks!

**Comment by stefankiesz** [comment]
Hi Farshid, can you please provide the master debug level logs by setting the following environment variable when running the sample: `AWS_KVS_LOG_LEVEL=2` and the viewer debug logs that are outputted on the test page.

Also, please verify that the RTSP stream has media by using something like a VLC player, then check the output from running the following command line pipeline.
```
gst-launch-1.0 uridecodebin uri=<your-uri> ! videoconvert ! autovideosink
```
There should be a window displaying your footage.

**Reply by farshidbakh** [reply]
Hello Stefan, Thank you so much. I got it working, I tried your suggestion on my Ubuntu machine:
gst-launch-1.0 uridecodebin uri=<your-uri> ! videoconvert ! autovideosink
and found out that I was missing some plugins. I installed them and now I get to see the IP camera stream on the Test Page. 
My mistake was that I had set the log level as AWS_KVS_LOG_LEVEL_VERBOSE=1 (my typo)  instead of AWS_KVS_LOG, that's why I was not seeing the error messages.   
I was wondering if there is an equivalent Java sdk to the C SDK that I was testing with? Do you know of any Java SDK?

Thank a lot,
Farshid

**Reply by stefankiesz** [reply]
@farshidbakh, I'm glad to hear that! And no, there is no Java equivalent to the C WebRTC SDK at the moment, but we are taking this into consideration for the future. Incase it fits your needs, we do have the [Java Producer SDK](https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-java) to ingest media for playback and analytics.

**Reply by farshidbakh** [reply]
Thanks Stefan, I am under impression that with the Java Producer SDK  I can send/relay stream to a StreamName on AWS. Can I send video stream using a SignalChannel? Thanks!

**Reply by stefankiesz** [reply]
That's right, it would stream to a KVS stream which can then be played back using HLS, DASH, or GetMedia. But no, it would not stream to a signaling channel.

**Reply by farshidbakh** [reply]
Thank you very much! Any idea when the Java SDK will be released supporting the Signal Channel?

**Reply by disa6302** [reply, marked as answer]
Hello @farshidbakh , 

We will take this in as a feature request to discuss internally, but we do not have plans of supporting this anytime soon.
